price-estimator/src/orderbook.rs

```diff
@@ -16,18 +16,14 @@ impl<T> Orderbook<T> {
             pricegraph: RwLock::new(Pricegraph::new(std::iter::empty())),
         }
     }
-
-    pub async fn get_pricegraph(&self) -> Pricegraph {
-        self.pricegraph.read().await.clone()
-    }
 }
 
 impl<T: StableXOrderBookReading> Orderbook<T> {
-    /// Recreate the pricegraph orderbook.
-    pub async fn update(&self) -> anyhow::Result<()> {
+    /// Creates a `Pricegraph` instance at the given batch ID.
+    async fn create_pricegraph_at_batch(&self, batch_id: BatchId) -> Result<Pricegraph> {
         let (account_state, orders) = self
             .orderbook_reading
-            .get_auction_data(current_batch_id() as u32)
+            .get_auction_data(batch_id.into())
```

Comment:
This is collecting all orders/balances as of the last block (inclusive) of this batchId, right?

Reply:
AFAIK this is inclusive (i.e. the solving batch, as per the `StableXOrderbookReading`).